Chart, background and formula change

Jerome81 opened the pull request:
This adds a line chart to the daily almanac containing the daily goal, words written and 7-day avg.
Makes the formula correct for the first 6 days.
Sets the background to gray for the backup copy.

Jerome81 committed:
Set background of backup to gray

This should make even less likely for beginners to accidentally edit the backup version and lose all changes.

Jerome81 committed:
Correct formula for the first 6 days of usage

The formula that gets written for the first 6 days is wrong (avgStart is negative). This change makes sure there is a legal formula for the first few days and should make life easier for new users.

Jerome81 committed:
Add a line chart to almanac

This change adds a line chart to the almanac email that contains the daily goal, the written words, and the 7-day average.

jamietr1 commented:
@Jerome81, I like the chart. It is similar to the dashboard I have created for my writing stats (I keep meaning to post the code for it), but conveniently contained in the email message. I do have a suggestion, however:

Could you add a setting to the spreadsheet, something like: "Include chart in email?" If the setting is true, include the chart, if it is false, don't include it. That way a user can decide for themselves if they want it, and change their preference at any time.

Jerome81 committed:
Make inclusion of chart configurable

The chart is included only if there is the following setting in the config tab:

Include Chart in Almanac | 1 | Set | Include a chart that contains the goal, total and 7-day average in the daily email

For backward compatibility, I added a method that takes a default value for the config and doesn't stop execution if it can't find the value in the spreadsheet. This value defaults to 0, so if a user copies this new version, he shouldn't see any change in behavior.

Jerome81 commented:
@jamietr1 Sure, no problem.
